.github/workflows/linters.yml

```diff
@@ -0,0 +1,161 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow
+
+name: "Linters"
+
+on:
+  push:
+    branches:
+      - "main"
+      - "v2.**"
+  pull_request: null
+# Add [skip ci] to commit message to skip CI.
+
+permissions: {}
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  byte_level:
+    name: "0️⃣ Byte-level"
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: "Checkout repository"
+        uses: "actions/checkout@v3"
+      - name: "Check file permissions"
+        run: |
+          test "$(find . -type f -not -path './.git/*' -executable)" = "./bin/create-command-test"
+      - name: "Find non-printable ASCII characters"
+        run: |
+          # ü for "Till Krüss"
+          ! LC_ALL=C.UTF-8 find . -type f -name '*.php' -print0 \
+            | xargs --null -- grep --perl-regexp --with-filename --line-number '[^ -~ü]'
+
+  syntax_errors:
+    name: "1️⃣ Syntax errors"
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: "Set up PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          php-version: "8.1"
+          coverage: "none"
+      - name: "Checkout repository"
+        uses: "actions/checkout@v3"
+      - name: "Install dependencies"
+        uses: "ramsey/composer-install@v2"
+        with:
+          dependency-versions: "highest"
+      - name: "Check source code for syntax errors"
+        run: "composer exec -- parallel-lint bin/ examples/ src/ tests/"
+
+  #unit_tests:
+    #name: "2️⃣ Unit and functional tests"
+    #needs:
+    #  - "byte_level"
+    #  - "syntax_errors"
+    #strategy:
+    #  #fail-fast: false
+    #  matrix:
+    #    php-version:
+    #      - "8.0"
+    #    dependencies:
+    #      - "lowest"
+    #      - "highest"
+    #runs-on: "ubuntu-latest"
+    #steps:
```

Comment:
May I include unit tests in this workflow?
That would stop CI when there is a 0️⃣ Byte-level or 1️⃣ Syntax error.

Reply:
@tillkruss How about this one? Merging two workflows for the above reason?

Reply:
No, let's keep those separate.

Reply:
🖖🏻